core/services/pipeline/runner.go

```diff
@@ -235,6 +245,7 @@ func init() {
 // overtimeContext returns a modified context for overtime work, since tasks are expected to keep running and return
 // results, even after context cancellation.
 func overtimeContext(ctx context.Context) (context.Context, context.CancelFunc) {
+	ctx = overtimeThresholds.ContextWithValue(ctx)
```

Comment:
Perhaps this should only apply to the deadline case? Because we have a longer default timeout otherwise